src/editors/containers/TextEditor/hooks.js

```diff
@@ -37,30 +37,35 @@ export const setupCustomBehavior = ({ openModal, setImage }) => (editor) => {
     onAction: openCodeEditor,
   });
   // add a custom simple inline code block formatter.
-  const toggleCodeFormatting = () => editor.formatter.toggle('code');
-  editor.ui.registry.addButton(tinyMCE.buttons.codeBlock, {
+  const setupCodeFormatting = (api) => {
+    editor.formatter.formatChanged(
+      'code',
+      (active) => api.setActive(active),
+    );
+  };
+  const toggleCodeFormatting = () => {
+    editor.formatter.toggle('code');
+    editor.undoManager.add();
+    editor.focus();
+  };
+  editor.ui.registry.addToggleButton(tinyMCE.buttons.codeBlock, {
     icon: 'sourcecode',
     tooltip: 'Code Block',
     onAction: toggleCodeFormatting,
-  });
-  const toggleBlockQuoteFormatting = () => editor.formatter.toggle('blockquote');
-  editor.ui.registry.addButton(tinyMCE.buttons.blockQuote, {
-    icon: 'quote',
-    tooltip: 'Block Quote',
-    onAction: toggleBlockQuoteFormatting,
```

Comment:
I removed this line about Block Quote because we are using TinyMCE's built-in Block Quote button and not a custom one.